framework/commercetools/lib/normalize.ts

```diff
@@ -0,0 +1,180 @@
+import type {
+  CommercetoolsProducts,
+  Product,
+  ProductVariant,
+  CommercetoolsProductVariant,
+  CommerceToolsProductPrice,
+  ProductPrice,
+} from '../types/product'
+import type {
+  Cart,
+  CommercetoolsCart,
+  CommercetoolsLineItems,
+  LineItem,
+} from '../types/cart'
+
+import type {
+  CommercetoolsBrands,
+  CommercetoolsCategory,
+  Category,
+  Brand,
+} from '../types/site'
+
+import { arrayToTree } from './array-to-tree'
+
+function normalizeVariants(
+  variants: CommercetoolsProductVariant[],
+  published: boolean
+): ProductVariant[] {
+  return variants.map((variant) => {
+    return {
+      id: variant.id,
+      options: [],
+      availableForSale: published,
+    }
+  })
+}
+
+function normalicePrice(price: CommerceToolsProductPrice): ProductPrice {
+  const value =
+    price.discounted && price.discounted.value
+      ? price.discounted.value.centAmount
+      : price.value.centAmount
+  return {
+    value,
+    currencyCode: price.value.currencyCode,
+    retailPrice: 0,
+    salePrice: 0,
+    listPrice: 0,
+    extendedListPrice: 0,
+    extendedSalePrice: 0,
+  }
+}
+
+export function normalizeProduct(data: CommercetoolsProducts): Product {
+  return {
+    id: data.id,
+    name: data.name.en,
+    description:
+      data.metaDescription && data.metaDescription.en
+        ? data.metaDescription.en
+        : 'No description',
+    slug: data.slug.en,
+    path: data.slug.en,
+    images: data.masterVariant.images,
+    variants: normalizeVariants(data.variants, data.published),
+    options: [],
+    price: normalicePrice(data.masterVariant.prices[0]),
```

Comment:
we can get the price field if exist, if not we get the first from the `prices` array